src/ops/direnv/mod.rs

```diff
@@ -15,8 +17,24 @@ pub fn main(project: &Project) -> OpResult {
     let mut shell_root = project.gc_root_path().unwrap();
     shell_root.push("build-0"); // !!!
 
+    // TODO: don’t start build/evaluation automatically, let the user decide
+    if let Ok(client) = client::ping(DEFAULT_READ_TIMEOUT).connect(
+        &::socket::path::SocketPath::from(::ops::get_paths()?.daemon_socket_file()),
+    ) {
+        client
+            .write(&Ping {
+                nix_file: project.expression().clone(),
+            })
+            .unwrap();
+    } else {
+        eprintln!("Uh oh, your lorri daemon is not running.");
+    }
+
+    // TODO: this needs to move to a daemon message
     if !shell_root.exists() {
```

Comment:
I wonder when under the circumstance:

 - if the daemon is up
 - and we ping successfully
 - and the shell root doesn't exist

if we should sleep until the initial evaluation is done?

Hmm maybe not, that might block editors... but perhaps the error below should be contextual, where in this situation we print a message saying it is not _yet_ ready.

Reply:
I never want to block the user by default, rather we should implement a command to wait for the next successful evaluation.

Reply:
Discussion in https://github.com/target/lorri/issues/99